### PR TITLE
Models module

### LINKED ISSUE #42: Add "models" folder
The utils folder is getting a bit bloated. I suggest that we move all code that is specialized to generating model Hamiltonians (i.e. hubbard, jellium, plane_wave, meanfield and possible molecular_data) to a different folder called "models". This will require changing imports in a lot of the code and also in docs/examples.rst and in the demo ipython notebook.

### PR DESCRIPTION
**Pull request by babbush**
Jarrod: DO NOT MERGE
* Significantly reordered repo to include "hamiltonians" folder
* Bumped version number as this will break all plugins
Fixes #42

### CODE REVIEW

File: src/openfermion/hamiltonians/_mean_field_dwave.py [hunk 1]
```diff
@@ -35,13 +35,13 @@
 """
 from __future__ import absolute_import
 
+from openfermion.hamiltonians._hubbard import up, down
```
Comment (jarrodmcc) [inline src/openfermion/hamiltonians/_mean_field_dwave.py]: I think perhaps more descriptive names for up and down are possible.  I almost feel like they should also be internal, e.g. _up, _down, so such generic names are not in the name space if someone imports * from hubbard.

File: src/openfermion/hamiltonians/_mean_field_dwave_test.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+
+class MeanfieldDwaveTest(unittest.TestCase):
+
+    def setUp(self):
+        self.tunneling = 2.
+        self.sc_gap = 2.
+
+    def test_two_by_two(self):
+
```
Comment (jarrodmcc) [inline src/openfermion/hamiltonians/_mean_field_dwave_test.py]: I know I write this review a lot, but why do we have no project standard on writing one-liners on generically what is being tested in a test.  It's all fine until a test breaks in 6 months and you have to spend all day figuring out what it was supposed to be testing...

File: src/openfermion/utils/_operator_utils.py [hunk 3]
```diff
@@ -103,57 +125,120 @@ def is_identity(operator):
     raise TypeError('Operator of invalid type.')
 
 
-def commutator(operator_a, operator_b):
-    """Compute the commutator of two operators.
+def _fourier_transform_helper(hamiltonian,
```
[the remark this answers is not in the document]
Comment (babbush) [reply src/openfermion/utils/_operator_utils.py]: I agree. Unfortunately, I am not totally clear myself. @Spaceenter wrote this code I believe so I would ask him to add the doc strings in a later pull request. But as this issue existed before my pull request, I would ask you to let it for go for this PR.

File: src/openfermion/hamiltonians/_mean_field_dwave.py (hunk 1 above)
Comment (babbush) [reply src/openfermion/hamiltonians/_mean_field_dwave.py]: Addressed

**Pull request merged**